src/parquet/file/reader.cc

```diff
@@ -41,17 +41,18 @@ namespace parquet {
 // ----------------------------------------------------------------------
 // RowGroupReader public API
 
-RowGroupReader::RowGroupReader(const SchemaDescriptor* schema,
-    std::unique_ptr<Contents> contents, MemoryAllocator* allocator)
-    : schema_(schema), contents_(std::move(contents)), allocator_(allocator) {}
+RowGroupReader::RowGroupReader(std::unique_ptr<Contents> contents)
+    : contents_(std::move(contents)) {}
 
 std::shared_ptr<ColumnReader> RowGroupReader::Column(int i) {
-  DCHECK(i < schema_->num_columns()) << "The RowGroup only has " << schema_->num_columns()
-                                     << "columns, requested column: " << i;
-  const ColumnDescriptor* descr = schema_->Column(i);
+  DCHECK(i < metadata()->num_columns()) << "The RowGroup only has "
+                                        << metadata()->num_columns()
+                                        << "columns, requested column: " << i;
+  const ColumnDescriptor* descr = metadata()->schema()->Column(i);
 
```

Comment:
The `RowGroupMetadata` is extended to include `SchemaDescriptor` to support this. Alternatively, we can modify the `contents` struct to include `std::unique_ptr<PageReader> Column(int i)` instead of  `std::unique_ptr<PageReader>  GetColumnPageReader(int i)`